rust/kernel/error.rs

```diff
@@ -104,3 +125,13 @@ impl From<AllocError> for Error {
         Error::ENOMEM
     }
 }
+
+/// Converts a kernel pointer to a [`Result`].
+///
+/// Kernel pointers can be used to store error values (see
+/// [`Error::from_kernel_ptr`]), therefore this function returns `Err` if `ptr
+/// indicates an error, otherwise returns `Ok(ptr)`.
+#[allow(dead_code)]
```

Comment:
Note: I have to make this as `#[allow(dead_code)]` because there is no real user since I separate this from my thread PR #109, this can be removed once we finish our discussion and have real user merged